elastic/kibana
# [App Search] Result settings form

## Summary

This PR adds the form to the Result Settings page of App Search.

With this PR, you should be able to view and modify Result Settings, but you will not yet be able to save them. That will be included in a subsequent PR.

The UI is visually migrated largely as-is. I migrated this *without* styles at first, then pulled in the minimum amount of styles I thought we needed to make this look decent. The HTML structure is more or less identical as well.

QA:
- To view settings that have already been made, you'll need to make them first in the ent-search app, as there is no way to save settings currently.
- There are 3 sections, each will only appear if the relevant data exists

    1. Text fields - Any field in the schema that has its type as "text". Can edit "Snippet" *and* "Raw" settings
    2. Non-text fields - Any field in the schema that has its type as something other than "text". Can *only* edit "Raw" setting.
    3. Disabled fields - If you are in a meta engine that has schema conflicts, this section will appear, as any field that is in a conflicted state is implicitly "disabled".

- There is a special "onBlur" handler, on the "Max size" fields. It ensures that if any non numeric value, or a value that is below 20, or above 1000, gets reset to a valid value.


__This PR__
![latest](https://user-images.githubusercontent.com/1427475/112531633-2faa0780-8d7e-11eb-8edb-20e48ccc0414.gif)


__ent-search__
<img width="1706" alt="ent-search" src="https://user-images.githubusercontent.com/1427475/112531618-29b42680-8d7e-11eb-9d10-18efbc7be32f.png">


### Checklist

Delete any items that are not applicable to this PR.


**Note**

This doesn't look *great* on mobile devices. I think I'll revisit this after we've added the rest of the page elements and think about how this would look best holistically.

### For maintainers
